hydro_cli_examples/examples/pn_counter/main.rs

```diff
@@ -0,0 +1,136 @@
+struct IncrementRequest {
+    tweet_id: u64,
+    likes: i32,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+enum GossipOrIncrement {
+    Gossip(HashMap<u64, (Vec<u32>, Vec<u32>)>),
+    Increment(u64, i32),
+}
+
+#[tokio::main]
+async fn main() {
+    let mut ports = hydroflow::util::cli::init().await;
+
+    let my_id: Vec<usize> = serde_json::from_str(&std::env::args().nth(1).unwrap()).unwrap();
+    let my_id = my_id[0];
+    let num_replicas: Vec<usize> = serde_json::from_str(&std::env::args().nth(2).unwrap()).unwrap();
+    let num_replicas = num_replicas[0];
+
+    let increment_requests = ports
+        .remove("increment_requests")
+        .unwrap()
+        .connect::<ConnectedBidi>()
+        .await
+        .into_source();
+
+    let query_responses = ports
+        .remove("query_responses")
+        .unwrap()
+        .connect::<ConnectedBidi>()
+        .await
+        .into_sink();
+
+    let to_peer = ports
+        .remove("to_peer")
+        .unwrap()
+        .connect::<ConnectedDemux<ConnectedBidi>>()
+        .await
+        .into_sink();
+
+    let from_peer = ports
+        .remove("from_peer")
+        .unwrap()
+        .connect::<ConnectedBidi>()
+        .await
+        .into_source();
+
+    let f1 = async {
+        #[cfg(target_os = "linux")]
+        loop {
+            let x = procinfo::pid::stat_self().unwrap();
+            println!("memory: {} bytes", x.rss * 1024 * 4);
+            tokio::time::sleep(Duration::from_secs(1)).await;
+        }
+    };
+
+    let df = hydroflow_syntax! {
+        next_state = merge()
+            -> fold::<'static>((0, HashMap::<u64, (Vec<u32>, Vec<u32>)>::new()), |(_, mut cur_state), goi| {
+                match goi {
+                    GossipOrIncrement::Gossip(gossip) => {
+                        for (counter_id, (pos, neg)) in gossip.iter() {
+                            let cur_value = cur_state.entry(*counter_id).or_insert((
+                                vec![0; num_replicas], vec![0; num_replicas]
+                            ));
+
+                            for i in 0..num_replicas {
+                                cur_value.0[i] = std::cmp::max(cur_value.0[i], pos[i]);
+                                cur_value.1[i] = std::cmp::max(cur_value.1[i], neg[i]);
+                            }
+                        }
+                    }
+                    GossipOrIncrement::Increment(counter_id, delta) => {
+                        let cur_value = cur_state.entry(counter_id).or_insert((
+                            vec![0; num_replicas], vec![0; num_replicas]
+                        ));
+                        if delta > 0 {
+                            cur_value.0[my_id] += delta as u32;
+                        } else {
+                            cur_value.1[my_id] += (-delta) as u32;
+                        }
+                    }
+                }
+
+                (context.current_tick(), cur_state)
```

Comment:
why is current tick being referenced?

Reply:
Because `fold::<'static>` replays (undid disabling that with a workaround in the Topolotree PR), this allows us to ignore the replay by following it with a simple `filter`.